File: specification/specs/x_pack/watcher/execute_watch/ActionExecutionState.ts
enum ActionExecutionState {
	awaits_execution = 0,
	checking = 1,
	execution_not_needed = 2,
	throttled = 3,
	executed = 4,
	failed = 5,
	deleted_while_queued = 6,
	not_executed_already_queued = 7
}

File: output/csharp/XPack/Watcher/ExecuteWatch/ActionExecutionState.cs

using System.Runtime.Serialization;

namespace Nest.XPack {

	public enum ActionExecutionState {
  
		[DataMember(Name = "awaits_execution")] AwaitsExecution,
		[DataMember(Name = "checking")] Checking,
		[DataMember(Name = "execution_not_needed")] ExecutionNotNeeded,
		[DataMember(Name = "throttled")] Throttled,
		[DataMember(Name = "executed")] Executed,
		[DataMember(Name = "failed")] Failed,
		[DataMember(Name = "deleted_while_queued")] DeletedWhileQueued,
		[DataMember(Name = "not_executed_already_queued")] NotExecutedAlreadyQueued
	}
}
